src/websocket/server.cc

```diff
@@ -262,25 +279,54 @@ future<websocket_parser::consumption_result_t> websocket_parser::operator()(
     return websocket_parser::stop(std::move(data));
 }
 
+future<> connection::handle_ping() {
+    // TODO
+    return make_ready_future<>();
+}
+
+future<> connection::handle_pong() {
+    // TODO
+    return make_ready_future<>();
+}
+
+
 future<> connection::read_one() {
     return _read_buf.consume(_websocket_parser).then([this] () mutable {
         if (_websocket_parser.is_valid()) {
             // FIXME: implement error handling
-            return _input_buffer.push_eventually(_websocket_parser.result());
+            switch(_websocket_parser.opcode()) {
+                // We do not distinguish between these 3 types.
+                case opcodes::CONTINUATION:
+                case opcodes::TEXT:
+                case opcodes::BINARY:
+                    return _input_buffer.push_eventually(_websocket_parser.result());
+                case opcodes::CLOSE:
+                    wlogger.debug("Received close frame.");
+                    /*
+                     * datatracker.ietf.org/doc/html/rfc6455#section-5.5.1
+                     */
+                    return close(true);
+                case opcodes::PING:
+                    return handle_ping();
+                    wlogger.debug("Received ping frame.");
+                case opcodes::PONG:
+                    wlogger.debug("Received pong frame.");
+                    return handle_pong();
+                default:
+                    // Invalid - do nothing.
```

Comment:
Is the opcode an integer that the client can send? If so, the client can send an invalid opcode, and we should probably do something about it (is there a way to return an error?)